core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -241,21 +242,22 @@ object DataWriteMethod extends Enumeration with Serializable {
  */
 @DeveloperApi
 case class InputMetrics(readMethod: DataReadMethod.Value) {
-
-  private val _bytesRead: AtomicLong = new AtomicLong()
+ 
+  @volatile @transient var bytesReadCallback: Option[() => Long] = None
 
   /**
    * Total bytes read.
    */
+  private val _bytesRead: AtomicLong = new AtomicLong()
   def bytesRead: Long = _bytesRead.get()
-  @volatile @transient var bytesReadCallback: Option[() => Long] = None
+  def incBytesRead(bytes: Long) = _bytesRead.addAndGet(bytes)
 
   /**
-   * Adds additional bytes read for this read method.
+   * Total records read.
    */
-  def addBytesRead(bytes: Long) = {
-    _bytesRead.addAndGet(bytes)
-  }
+  def recordsRead: Long = _recordsRead.get() 
+  private val _recordsRead: AtomicLong = new AtomicLong()
```

Comment:
This will only ever be modified by a single thread, right? So better to make it volatile than atomic long?